### PR TITLE
Feature/brother ql

### PR DESCRIPTION
**Pull request by poelzi**
###### Motivation for this change

Allows to print label with the Brother QL printer line. Tested and works with a QL-800 printer on Linux.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/brother_ql/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "d77fdfdf0ab61ca45b9599fc3270fe98faa16d992e4ce098c66ac7f722787b87";
+  };
+
+  propagatedBuildInputs = [ future packbits pillow pyusb ];
+
+  meta = with stdenv.lib; {
+    description = "Python package for the raster language protocol of the Brother QL series label printers (QL-500, QL-550, QL-570, QL-700, QL-710W, QL-720NW, QL-800, QL-820NWB, QL-1050 and more)";
```
Comment (dotlambda) [inline pkgs/development/python-modules/brother_ql/default.nix]: Too long

File: pkgs/development/python-modules/brother_ql/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    homepage = https://github.com/pklaus/brother_ql;
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/brother_ql/default.nix]: Maintainer required.

File: pkgs/development/python-modules/packbits/default.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "bc6b370bb34e04ac8cfa835e06c0484380affc6d593adb8009dd6c0f7bfff034";
+  };
+
+  meta = with stdenv.lib; {
+    description = "PackBits encoder/decoder for Python";
+    homepage = https://github.com/psd-tools/packbits;
+    license = [ licenses.bsd2 ];
+    maintainers = with maintainers; [ ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/packbits/default.nix]: Maintainer required

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -232,6 +232,8 @@ in {
 
   breathe = callPackage ../development/python-modules/breathe { };
 
+  brother_ql = callPackage ../development/python-modules/brother_ql { };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Is `brother_ql` really a Python module, seems more like a CLI tool only. In that case, it doesn't belong in `pythonPackages` and should use `buildPythonApplication`.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please add a `pythonPackages.` prefix to the appropriate commit messages.

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (poelzi) [reply pkgs/top-level/python-packages.nix]: It is a library and application. There is a brother_ql_web program that uses brother_ql as a dependency. It has however also some scripts to create labels.

Comment (dotlambda) [reply pkgs/top-level/python-packages.nix]: Okay, then it belongs into `pythonPackages`.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please create three commits where one adds yourself to `maintainer-list.nix` and one for each package.

Also note that there's a merge conflict. You should rebase on master.

**Comment by mmahut** [comment]
Any updates on this pull request, please?

**Comment by Ekleog** [comment]
@poelzi Are you still interested in moving this forward?

**Pull request closed without merging**